exercises/template-driven-forms/README.md

```diff
@@ -2,4 +2,75 @@
 
 Create a template driven form with the following behavior
 
-![](./preview.gif)
\ No newline at end of file
+![](./preview.gif)
+
+## Step 1
+
+Using the `angular-cli` create a new project called `template-driven-forms` inside of the `training` folder.
+
+## Step 2
+
+Create a Typescript `interface` called `Person` that captures the data model of the form in a file called `person.model.ts`
+
+## Step 3
+
+Modify the `app.component.ts` to import the `Person` created in Step 2 and declare a property of that type on your `AppComponent`.  Also import the `NgForm` directive.
+
+## Step 4
+
+Declare properties (of the appropriate types, or appropriately initialized) in `AppComponent` to store the possible options of the radio and select controls of the form as well as a property `isVisible` to store if the display data is available or not.
+
+## Step 5
+
+Modify the `app.component.html` template file to contain the HTML content of the form (with a container `form` element, with `NgForm` directive, and template variable allowing the `form` to be referenced in the HTML) and continaing `label`s, appropriate `input` and `select` elements, and a `submit` `button`, add the `NgModel` directive to the `input` elements)
+
+## Step 6
+
+Modify the `input` elements to add template variables to each so that they can be referenced in the HTML.  Also add the `required` attribute to all of them.
+
+## Step 7
+
+For the country `select` element put an `option` (set with `selected` and `disabled` attributes) for the '--Select a Country--' title option.  Use the `NgFor` directive to add the set options for the possible countries based on the appropriate initialized property that was added to `AppComponent`.  Optionally also use `NgFor` to define the radio options for the gender `input` element.
```

Comment:
we could, but currently the solution doesn't use it.  Should I modify the existing solution to use it?  Currently the solution file has:

```html
<label>
      Male:
      <input type="radio" name="gender" value="Male" 
        ngModel #gender="ngModel" required />
    </label>
    <label>
      Female:
      <input type="radio" name="gender" value="Female" 
        ngModel #gender="ngModel" required />
    </label>
    
    <!--<label *ngFor="let gender of options.genders">
      {{ gender }} 
      <input type="radio" name="gender" 
        [value]="gender" [ngModel]="person.gender" #gender="ngModel" required />
    </label>-->
```